sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -461,6 +502,77 @@ def convert_feature(fx):
         )
     features = tmp_features
 
+    if categorical_features is None:
+        is_categorical = [
+            (False,) if len(fxs) == 1 else (False, False) for fxs in features
+        ]
+    else:
+        categorical_features = np.asarray(categorical_features)
+        if categorical_features.dtype.kind == "b":
+            if categorical_features.size != n_features:
+                raise ValueError(
+                    "When `categorical_features` is a boolearn array-like, "
+                    "the array should be of shape (n_features,). Got "
+                    f"{categorical_features.size} elements while `X` contains "
+                    f"{n_features} features."
+                )
+            is_categorical = [
+                tuple(categorical_features[fx] for fx in fxs) for fxs in features
+            ]
+        elif categorical_features.dtype.kind in ("i", "O", "U"):
+            categorical_features_idx = [
+                _get_feature_index(cat, feature_names=feature_names)
+                for cat in categorical_features
+            ]
+            is_categorical = [
+                tuple([idx in categorical_features_idx for idx in fxs])
+                for fxs in features
+            ]
+        else:
+            raise ValueError(
+                "Expected `categorical_features` to be an array-like of boolean,"
+                f" interger, or string. Got {categorical_features.dtype} instead."
```

Comment:
nit: the `dtype` here is whatever `np.asarray` has decided though, not the original one. Not sure if we care.